src/solana/SolanaEngine.ts

```diff
@@ -361,33 +369,133 @@ export class SolanaEngine extends CurrencyEngine<
       throw new Error('makeSpend Missing publicAddress')
     if (nativeAmount == null) throw new NoAmountSpecifiedError()
 
-    const nativeNetworkFee = this.feePerSignature
+    const isTokenTx = tokenId != null
 
-    const balanceSol = this.walletLocalData.totalBalances[this.chainCode] ?? '0'
     let totalTxAmount = '0'
-    totalTxAmount = add(nativeAmount, nativeNetworkFee)
-    if (gt(totalTxAmount, balanceSol)) {
-      throw new InsufficientFundsError()
-    }
-    // Create Solana transaction
-    const payerPublicKey = new PublicKey(this.base58PublicKey)
+    let nativeNetworkFee = this.feePerSignature
+    let parentNetworkFee: string | undefined
+    const balance = this.walletLocalData.totalBalances[currencyCode] ?? '0'
+
+    // pubkeys
+    const PAYER = new PublicKey(this.base58PublicKey)
+    const PAYEE = new PublicKey(publicAddress)
+    const TOKEN_PROGRAM_ID = new PublicKey(this.networkInfo.tokenPublicKey)
+    const ASSOCIATED_TOKEN_PROGRAM_ID = new PublicKey(
+      this.networkInfo.associatedTokenPublicKey
+    )
```

Comment:
These variable names don't fit our convention. Is there some reason?

Reply:
No good reason, will fix